soc/nordic/common/dmm.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+int dmm_dma_buffer_out_prepare(const struct device *dev, void const * user_buffer, void ** buffer_out)
+{
+    // Get memory region that specified device can perform DMA transfers from, using devicetree
+
+    // Check if provided user buffer is already in correct memory region
+    // If yes, assign buffer_out to user_buffer
```

Comment:
First, check if the buffer is aligned and padded to cache lines, or if it is in non-cacheable memory. Only if these conditions are met assign it to user_buffer.

Reply:
added